### PR TITLE
Make user preference selection look like dropdown menus instead of links

### PR DESCRIPTION
**Pull request by braunjj**
## Summary & Motivation
Right now these preference selectiosn look like links and it's not obvious that I can click them to make a different selection.

### Old:
image

### New:
image

## How I Tested These Changes
Loaded up the UI

### CODE REVIEW

**Comment by braunjj** [comment]
@hellendag I'd really like to make these right aligned but I can't figure out how to do that with the metadata table component. Any tips?

File: js_modules/dagster-ui/packages/ui-core/src/app/UserSettingsDialog.tsx [hunk 1]
```diff
@@ -123,15 +125,15 @@ const UserSettingsDialogContent = ({onClose, visibleFlags}: DialogContentProps)
               {
                 key: 'Timezone',
                 value: (
-                  <Box margin={{bottom: 4}}>
+                  <Box>
```
Comment (hellendag) [inline js_modules/dagster-ui/packages/ui-core/src/app/UserSettingsDialog.tsx:128]: Should probably be able to remove all of these `<Box>` wrappers now.

**Review by hellendag — APPROVED** [review]
> @hellendag I'd really like to make these right aligned but I can't figure out how to do that with the metadata table component. Any tips?

Given that the buttons are different widths, could that end up looking a little weird with the left sides of the buttons not aligned?

I think you'd probably need to keep the `Box` wrappers on each of the select components, and use `justifyContent: 'flex-end'`, something like that.

**Comment by hellendag** [comment]
Oh also, the vertical alignment of the keys and the buttons is a little weird, with the text aligning to the top of the button and not centered. Maybe using `<Box margin={{top: 4}}>` or something on `label` properties would help align them better?

**Pull request merged**